src/solvers/smt2_incremental/smt_bitvector_theory.cpp

```diff
@@ -0,0 +1,42 @@
+// Author: Diffblue Ltd.
+
+#include <solvers/smt2_incremental/smt_bitvector_theory.h>
+
+#include <util/invariant.h>
+
+static void validate_bitvector_predicate_arguments(
+  const smt_termt &left,
+  const smt_termt &right)
+{
+  const auto left_sort = left.get_sort().cast<smt_bit_vector_sortt>();
+  INVARIANT(left_sort, "Left operand must have bitvector sort.");
+  const auto right_sort = right.get_sort().cast<smt_bit_vector_sortt>();
+  INVARIANT(right_sort, "Right operand must have bitvector sort.");
+  INVARIANT(
+    left_sort->bit_width() == right_sort->bit_width(),
+    "Left and right operands must have the same bit width.");
+}
+
+#define SMT_BITVECTOR_THEORY_PREDICATE(the_identifier, the_name)               \
+  void smt_bit_vector_theoryt::the_name##t::validate(                          \
+    const smt_termt &left, const smt_termt &right)                             \
+  {                                                                            \
+    validate_bitvector_predicate_arguments(left, right);                       \
+  }                                                                            \
+                                                                               \
+  smt_sortt smt_bit_vector_theoryt::the_name##t::return_sort(                  \
+    const smt_termt &, const smt_termt &)                                      \
+  {                                                                            \
+    return smt_bool_sortt{};                                                   \
+  }                                                                            \
```

Comment:
Do we need the arguments here?

Reply:
No. Which is why they are unnamed. It is more straightforward to leaved them unused than it is to make the template code pass them only in some circumstances.